.ci/docs/build-wheel-matrix-ci.md

```
@@ -0,0 +1,529 @@
+# Stage 2: Default stage that builds and generates wheels
+FROM wheel-base
+# ... (NIXL build and wheel generation)
+```
+
+**Base Image**: `harbor.mellanox.com/nixl/$arch/cuda:12.8-devel-manylinux--25.03`
+
+**This image is from**: `gitlab-master.nvidia.com:5005/dl/dgx/cuda:12.8-devel-manylinux--25.03`
+
+### Stage Usage Patterns
+
+#### CI Pipeline Usage (Base Stage Only)
```

markdownlint-cli2 (0.22.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)